ssha/config.py

```diff
@@ -125,6 +129,35 @@ def get(key, default=None):
     return render(value)
 
 
+def generate_key():
+    """
+    Generates and adds a key to the SSH agent.
+
+    """
+    key = rsa.generate_private_key(
+        backend=crypto_default_backend(),
+        public_exponent=65537,
+        key_size=4096
+    )
+    private_key = key.private_bytes(
+        crypto_serialization.Encoding.PEM,
+        crypto_serialization.PrivateFormat.PKCS8,
+        crypto_serialization.NoEncryption())
+    public_key = key.public_key().public_bytes(
+        crypto_serialization.Encoding.OpenSSH,
+        crypto_serialization.PublicFormat.OpenSSH
+    )
+
+    with tempfile.NamedTemporaryFile() as temp_key:
+        os.chmod(temp_key.name, 0o600)
+        with open(temp_key.name, 'w') as open_file:
+            open_file.write(private_key)
+        os.system('ssh-add -t 60 {}'.format(temp_key.name))
+        temp_key.close()
```

Comment:
You don't need to close it. It is automatically closed at the end of the `with` block.